File: projects/unity/Tenlastic SDK/Assets/Tenlastic/Scripts/Jwt.cs
using JWT.Builder;
using System;
using UnityEngine;

namespace Tenlastic {
    public class Jwt {

        [Serializable]
        public struct Payload {
            public int exp;
            public int iat;
            public string jti;
            public UserModel user;
        }

        public bool isExpired {
            get {
                DateTimeOffset now = DateTimeOffset.Now;
                DateTimeOffset exp = DateTimeOffset.FromUnixTimeSeconds(payload.exp);

                return DateTimeOffset.Compare(exp, now) < 0;
            }
        }
        public Payload payload {
            get {
                return _payload;
            }
        }
        public string value {
            get {
                return _value;
            }
            set {
                _value = value;

                string json = new JwtBuilder().Decode(value);
                _payload = JsonUtility.FromJson<Payload>(json);
            }
        }

        private Payload _payload;
        private string _value;

        public Jwt(string value) {
            this.value = value;
        }

    }
}

File: nodejs/modules/http/src/jwt.ts
import * as jwt from 'jsonwebtoken';

import { UserModel } from './models/user';

export interface JwtPayload {
  exp?: number;
  jti?: string;
  user?: UserModel;
}

export class Jwt {
  public get isExpired() {
    return Date.now() >= this.payload.exp * 1000;
  }
  public get payload() {
    return this._payload;
  }
  public get value() {
    return this._value;
  }
  public set value(value: string) {
    this._value = value;
    this._payload = jwt.decode(value) as JwtPayload;
  }

  private _payload: JwtPayload;
  private _value: string;

  constructor(value: string) {
    this.value = value;
  }
}
